lab_5_scrapper/scrapper.py

```diff
@@ -13,70 +57,138 @@ def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+        config_dto = self._extract_config_content()
+        self._seed_urls = config_dto.seed_urls
+        self._num_articles = config_dto.total_articles
+        self._headers = config_dto.headers
+        self._encoding = config_dto.encoding
+        self._timeout = config_dto.timeout
+        self._should_verify_certificate = config_dto.should_verify_certificate
+        self._headless_mode = config_dto.headless_mode
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as file:
+            info = json.load(file)
+
+        config_dto = ConfigDTO(info['seed_urls'],
+                               info['headers'],
+                               info['total_articles_to_find_and_parse'],
+                               info['encoding'],
+                               info['timeout'],
+                               info['should_verify_certificate'],
+                               info['headless_mode'])
+
+        return config_dto
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as file:
+            info = json.load(file)
+
+        seed_urls = info['seed_urls']
+        headers = info['headers']
+        total_articles_to_find_and_parse = info['total_articles_to_find_and_parse']
+        encoding = info['encoding']
+        timeout = info['timeout']
+        verify_certificate = info['should_verify_certificate']
+        headless_mode = info['headless_mode']
```

Comment:
is it necessary to define all these variables here? you have the same class attributes fields - can you use them instead?

Reply:
fixed